web/modules/custom/nidirect_webforms/js/prisonbooking.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @file
+ * JQuery for prison visit booking webform.
+ */
+
+(function($, Drupal, once) {
+
+  Drupal.behaviors.prisonVisit = {
+    attach: function (context, settings) {
+
+      // The number of additional adult visitors determines
+      // the number of additional child visitors.
+
+      const $additionalAdults = $(once('prisonVisitAdditionalAdults', '[name="additional_visitor_adult_number"]', context));
+      const $additionalChildren = $(once('prisonVisitAdditionalChildren', '[name="additional_visitor_child_number"]', context));
+
+      if ($additionalAdults.val() > 0) {
+        $('option', $additionalChildren)
+          .slice(($(this).length - 1) - parseInt($additionalAdults.val()))
+          .attr('disabled', true);
+      }
+
+      $additionalAdults.on('change', function(e) {
+        let numAdults = $(this).val() ?? 0;
+
+        if (numAdults > 0) {
+          $('option', $additionalChildren)
+            .removeAttr('disabled')
+            .slice(($(this).length - 1) - numAdults)
+            .attr('disabled', true)
+            .prop('selected', false);
+        } else {
+          $('option', $additionalChildren).removeAttr('disabled')
+        }
+
+        if ($('option:selected', $additionalChildren).is(':disabled')) {
+          $('option').not(':disabled').last().prop('selected', true);
+        }
+      });
+
+
+      // Visit time slots grouped by week in details containers.
+      const $weekSlots = $(once('pvSlots', '[data-webform-key^="slots_week"]', context));
+
+      // One week of slots only? Force open the details container.
+      if ($weekSlots.length === 1) {
+        $weekSlots.prop("open", true);
+        $('summary', $weekSlots)
+          .prop('aria-expanded', true)
+          .prop('aria-pressed', true);
+      }
+
+      // Visit time slots preferences...
+
+      // Add aria-live region for announcing time slot preferences.
+      const $preferredStatus = $('<div role="region" class="visually-hidden" aria-live="polite" />');
+      $preferredStatus.append($('<h2 id="timeslots-announce-title" />'));
+      $preferredStatus.append($('<div id="timeslots-announce-description" />'));
+
+      $(once('pvPreferredStatus', '[data-webform-key="visit_preferred_day_and_time"]', context)).prepend($preferredStatus);
+
+      // Limit timeslots that can be selected and
+      // keep track of preferred time slots.
+      let $timeSlots = $('input[type="checkbox"]', $weekSlots);
+      $timeSlots.attr('aria-controls', 'timeslots-announce');
+
+      let preferredSlots = [];
+      let pvTypeId = 'F';
+      let timeSlotLimit = 3;
+
+      if ($timeSlots.length) {
+
+        // Determine the time slot limit. By default it's 3,
+        // but for virtual visits it's 5.
+        if (settings.prisonVisitBooking['booking_ref'] !== null) {
+          pvTypeId = settings.prisonVisitBooking.booking_ref.visit_type_id;
+          timeSlotLimit = (pvTypeId === 'V') ? 5 : 3;
```

Comment:
I've put quite a few constants in a config file. So have added a visit_type_time_slot_limit setting and retrieving that instead of hardcoding ...

```
      // Default visit type is 'F' for face to face.
      let pvTypeId =  'F';
      // By default, user can choose 3 timeslots.
      let pvTimeSlotLimit = 3;

      // Get visit type from the booking reference and get
      // the timeslot limit for that visit type from settings.
      if (settings.prisonVisitBooking['booking_ref'] !== null) {
        pvTypeId = settings.prisonVisitBooking.booking_ref.visit_type_id;
        pvTimeSlotLimit = settings.prisonVisitBooking.visit_type_time_slot_limit[pvTypeId];
      }
```